universal-login-sdk/src/core/services/WalletService.ts

```diff
@@ -148,9 +148,11 @@ export class WalletService {
       contractAddress: connectingWallet.contractAddress,
       key: connectingWallet.publicKey,
     };
-    return new Promise((resolve, reject) => {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
       const setWallet = this.setWallet.bind(this);
-      const unsubscribe = this.sdk.subscribe('KeyAdded', filter, () => {
+      const addKeyEvent = await this.sdk.walletContractService.getEventNameFor(connectingWallet.contractAddress, 'KeyAdded');
```

Comment:
Maybe we shoud move line 154 to 151 and than remove async from promise executor?

Reply:
✅